### ISSUE TITLE
regarding to training on facades dataset with cycleGAN

### ISSUE THREAD

**Issue by zyzhang1130** [junyanz/pytorch-CycleGAN-and-pix2pix#616]
Hi, I have completed 200 bathes of training with batch size=4 on facades datasets with cycleGAN. However, the fake images look still very distorted compared to the real images (despite rec looks find and I'm also wondering why is that the case). What can I do to improve the quality of the fake images?

Thank you.

**Comment by junyanz** [comment]
Could you use batchsize=1? I haven't tested the batchsize=4 case.

**Comment by Makhaon** [comment]
I have trained zebras dataset with batch=3 and seems that it works fine here.

**Comment by xml94** [comment]
@junyanz @Makhaon 
Could you give some explainment for this batch size issue.

I am facing same question in pix2pix algorithm. If I improve the batch_size up to 16 or 32 in facade training dataset, the results would be very blurring and the images' qualities would be bad, for example the window's glass would be not clear.

Same situation if I change the instance normalziation to batch normalization as well as the big batch size.

If I want to improve the batch size, what I need to do?

Thanks very much.

**Comment by junyanz** [comment]
@phillipi Instancenorm with a big batch size should be fine? Also, make sure that you use the same batch_size during training and test. By default, the test code will use batch_size=1. You can change it in the code.

**Comment by phillipi** [comment]
Yeah, I think instance norm with a big batch size should be fine. Instance norm itself shouldn't be affected by batch size.
